### DISCUSSION TITLE
Laravel Collection helper for digging through deep properties

### DISCUSSION THREAD

**Discussion by ZebTheWizard** [laravel/framework#45345] [Ideas]
Wish there was a more elegant way to pluck deep properties when working with large datasets.

Recently, we needed to calculate the average dpi for a collection of images corresponding to a product.

Normally we try to write something like this because it's easier to read than loops

```
collect($variations)
->pluck('images')
->flatten()
->pluck('attributes')
->flatten()
->pluck('dpi')
->average()
```

Wish we could simplify the above to something like this:

```
collect($variations)
->dig('images.attributes.dpi')
->average()
```

Always copying and pasting this snippet between projects so I quickly write out common calculations

**Comment by henzeb** [comment]
Pluck supports nested values. https://laravel.com/docs/9.x/collections#method-pluck
